File: lifeFull/SpringBootLifeSamadhan/src/main/java/com/lifesamadhan/api/model/CustomerProfile.java
package com.lifesamadhan.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Entity
@Table(name = "customer_profiles")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CustomerProfile {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotNull(message = "UserId is required")
    @Min(value = 1, message = "UserId must be a valid value")
    @Column(name = "user_id", nullable = false, unique = true)
    private Long userId;
    
    @Column(name = "location_id")
    private Long locationId;
    
    @Column(length = 255)
    private String address;
    
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "location_id", insertable = false, updatable = false)
    private Location location;
}

File: LifeSamadhan/Backend/Models/CustomerProfile.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeSamadhan.API.Models
{
    [Table("customerprofiles")]
    public class CustomerProfile
    {
        [Key]
        [Column("Id")]
        public long Id { get; set; }

        [Column("LocationId")]
        public long LocationId { get; set; }

        [StringLength(500, ErrorMessage = "Address is too long")]
        [Column("Address")]
        public string Address { get; set; } = string.Empty;

        [Required(ErrorMessage = "UserId is required")]
        [Column("UserId")]
        public long UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public User? User { get; set; }
    }
}